apps/studio/components/interfaces/Database/Roles/CreateRolePanel.tsx

```diff
@@ -50,113 +76,124 @@ const CreateRolePanel = ({ visible, onClose }: CreateRolePanelProps) => {
     })
   }
 
+  const handleClose = () => {
+    onClose()
+    form.reset()
+  }
+
   return (
     <SidePanel
       size="large"
       visible={visible}
       header="Create a new role"
       className="mr-0 transform transition-all duration-300 ease-in-out"
       loading={false}
-      onCancel={onClose}
+      onCancel={handleClose}
       customFooter={
         <div className="flex w-full justify-end space-x-3 border-t border-default px-3 py-4">
-          <Button
-            size="tiny"
-            type="default"
-            htmlType="button"
-            onClick={onClose}
-            disabled={isCreating}
-          >
-            Cancel
-          </Button>
-          <Button
-            size="tiny"
-            type="primary"
-            htmlType="button"
-            disabled={isCreating}
-            loading={isCreating}
-            onClick={() => submitRef?.current?.click()}
-          >
-            Create role
-          </Button>
+          <FormActions
+            form={formId}
+            isSubmitting={isCreating}
+            hasChanges={form.formState.isDirty}
+            handleReset={() => form.reset()}
+          />
         </div>
       }
     >
-      <Form
-        validateOnBlur
-        id={formId}
-        initialValues={initialValues}
-        validate={validate}
-        onSubmit={onSubmit}
-      >
-        {({ isSubmitting, handleReset, values, initialValues }: any) => {
-          return (
-            <div>
-              <FormSection
-                header={
-                  <FormSectionLabel className="lg:!col-span-4">Role Configuration</FormSectionLabel>
-                }
-              >
-                <FormSectionContent loading={false} className="lg:!col-span-8">
-                  <Input id="name" label="Name" />
-                </FormSectionContent>
-              </FormSection>
+      <Form_Shadcn_ {...form}>
+        <form
+          id={formId}
+          className="grid gap-6 w-full px-8 py-8"
+          onSubmit={form.handleSubmit(onSubmit)}
+        >
+          <FormField_Shadcn_
+            control={form.control}
+            name="name"
+            render={({ field }) => (
+              <FormItem_Shadcn_ className="grid gap-2 md:grid md:grid-cols-12 space-y-0">
+                <FormLabel_Shadcn_ className="flex flex-col space-y-2 col-span-4 text-sm justify-center text-foreground-light">
+                  Name
+                </FormLabel_Shadcn_>
+                <FormControl_Shadcn_ className="col-span-8">
+                  <Input_Shadcn_ {...field} className="w-full" />
+                </FormControl_Shadcn_>
+                <FormMessage_Shadcn_ className="col-start-5 col-span-8" />
+              </FormItem_Shadcn_>
+            )}
+          />
+          <div className="grid gap-2 mt-4 md:grid md:grid-cols-12">
+            <div className="col-span-4">
```

Comment:
these tailwind classes could be constants (or maybe even components) so we can reuse them elsewhere.

or maybe we make custom classes in which we can control the size / col width of these layouts from 1 place.

Reply:
I was thinking the exact same thing (classes as string constants). I'm still not fully convinced on having some classes set as default on FormLabel, FormItem, FormControl, FormMessage components.

Reply:
same same same 
I say we try and tackle this early next year. Adding to the 2024 list from yesterday.